tqftpserv.c

```diff
@@ -522,24 +555,44 @@ static int handle_writer(struct tftp_client *client)
 	}
 
 	payload = len - 4;
+	buf += 4;
 
-	ret = write(client->fd, buf + 4, payload);
-	if (ret < 0) {
-		/* XXX: report error */
-		printf("[TQFTP] failed to write data\n");
+	/* Check if the destination buffer has enough space */
+	if (client->cur_buf_loc + payload > client->rw_buf_size) {
+		printf("[TQFTP] Buffer overflow detected\n");
+		tftp_send_error(client->sock, 4, "Buffer overflow");
 		return -1;
 	}
 
-	tftp_send_ack(client->sock, block);
+	/* Copy the data to the destination buffer */
+	memcpy(client->rw_buf+client->cur_buf_loc, buf, payload);
+	client->rw_buf_size -= payload;
```

Comment:
The size doesn't decrease when you add more data. Please rename `rw_buf_size` accordingly.